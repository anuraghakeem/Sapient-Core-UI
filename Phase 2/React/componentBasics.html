<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Component Basics</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.6.2/react.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/15.6.2/react-dom.js"></script>
</head>

<body>
    <divn id="container">

    </div>
    <script>
        //Create Component
        //Instantiating the component

        class BasicComponent extends React.Component {

            render() {
                return React.createElement('h1', {}, "hello component");
            }
        }

        var basicCompInstance = React.createElement(BasicComponent);
        ReactDOM.render(basicCompInstance, document.getElementById('container'));
    </script>
</body>

</html>