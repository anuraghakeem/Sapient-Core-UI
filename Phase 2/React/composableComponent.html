<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Component Basics With JSX</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.6.2/react.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/15.6.2/react-dom.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/0.14.0-alpha1/JSXTransformer.js"></script>
</head>

<body>
    <divn id="container">

    </div>
    <script type="text/jsx">
        //1. Create Component
        //2.    Instantiating the component

        class FormComponent extends React.Component {

            render() {
                return  <form>
                        <input type="text" placeholder="Enter component"/>
                        <input type="button" value="Add" />
                        </form>
            }
        }

        class CommentComponent extends React.Component {

            render() {
                return <ul>
                            <li> React is a library</li>
                            <li> React is a blue</li>
                        </ul>
            }
        }

        class BlogComponent extends React.Component {

            render() {
                return <div>
                            <FormComponent/>
                            <CommentComponent/>
                        </div>
            }
        }

        //Using multiple instances for REUSABILITY
        var basicCompInstance = <div><BlogComponent/>
                                <BlogComponent/></div>
        ReactDOM.render(basicCompInstance, document.getElementById('container'));
    </script>
</body>

</html>