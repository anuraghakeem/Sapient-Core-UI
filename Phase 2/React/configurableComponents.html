<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Configurable Components</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.6.2/react.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/15.6.2/react-dom.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/0.14.0-alpha1/JSXTransformer.js"></script>
</head>

<body>
    <divn id="container">

        </div>
        <script type="text/jsx">
        //1. Create Component - CourseComponent ----> <h1>React</h1> 
        //2. Create component - ListOfCourses   ----> 5 course instances

        class CourseComponent extends React.Component {

            render() {
                return  <span>
                        <h1>{this.props.coursename}</h1>
                        <h3>Duration :</h3>{this.props.duratrion}
                        </span>
            }
        }
        
        class ListOfCoursesComponent extends React.Component {
            constructor(){
                super();
                this.courseOne="React";
                this.courseTwo="Angular";
                this.courseThree="Node";
            }
            render() {
                return  <div>
                        <CourseComponent coursename= {this.courseOne} duratrion="3 days"/>
                        <CourseComponent coursename={this.courseTwo} duratrion="4 days"/>
                        <CourseComponent coursename={this.courseThree} duratrion="2 days"/>
                        {/*  CourseComponent coursename="Mongo" duratrion="6 days"/>
                        <CourseComponent coursename="Express" duratrion="2 days"/>   */}
                    </div>
            }
        }

        var ListOfCoursesComponentInstance = <ListOfCoursesComponent/>
        ReactDOM.render(ListOfCoursesComponentInstance, document.getElementById('container'));
    </script>
</body>

</html>