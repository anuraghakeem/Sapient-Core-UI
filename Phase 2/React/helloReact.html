<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>hello React</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.6.2/react.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/15.6.2/react-dom.js"></script>
    
</head>

<body>
    <divn id="container">

        </div>
    <script>
        //Create React Element! React.createElement(<tag>eg:'h1'   ,   ids and class eg: {id:"container"}}   ,   
        //child   eg:"HELLO")
        var h1Element = React.createElement('h1', null, "HELLO");
        var pElement = React.createElement('p', {id:"firstPara"}, "This is a para"); 
        var fragment=[h1Element,pElement];
        var sectionElement = React.createElement('section', {}, fragment);
        ReactDOM.render(sectionElement, document.getElementById('container'));
    </script>
</body>

</html>
