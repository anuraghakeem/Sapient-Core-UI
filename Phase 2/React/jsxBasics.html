<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>JSX with React Intro</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.6.2/react.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/15.6.2/react-dom.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/0.14.0-alpha1/JSXTransformer.js"></script>

    <style>
        .header{
            color: green;
        }
    </style>
</head>

<body>
    <divn id="container">

        </div>
        <script type="text/jsx">
        var sectionElement = <section>
                                <h1 className="header"> hello JSX!</h1>
                                <p> This is a paragraph1</p>
                                {/*  Multi line comment
                                    works on multiple lines*/}
                                <p> This is a paragraph2</p>
                            </section>
        ReactDOM.render(sectionElement, document.getElementById('container'));
    </script>
</body>

</html>