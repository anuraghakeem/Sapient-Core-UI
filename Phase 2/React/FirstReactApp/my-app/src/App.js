import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
class App extends React.Component {
  render() {
    return (
      <div className="App">
        <h1>App using Create React App</h1>
      </div>
    );
  }
}
export default App;
