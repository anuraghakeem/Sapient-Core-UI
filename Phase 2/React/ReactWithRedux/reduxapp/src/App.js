import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Provider } from "react-redux";
import { ShoppingCart } from "./components/shoppingcart.component";
import { store } from "./store/store";
import {Main} from "./connect.js"

function App() {
  return (
    <Provider store={store}>
      <Main />
    </Provider>
  );
}

export default App;
