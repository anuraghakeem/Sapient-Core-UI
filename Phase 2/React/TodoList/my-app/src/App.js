import React from 'react';
import logo from './logo.svg';
import './App.css';
import {Provider} from 'react-redux';
import {store} from './store/store'
import {Main} from './connect'
import { Route, BrowserRouter,Switch, Link } from 'react-router-dom';

function App() {
  return (
    <Provider store={store}>
       <BrowserRouter>
          <Main/>
        </BrowserRouter>
  
      </Provider>
  );
}

export default App;
