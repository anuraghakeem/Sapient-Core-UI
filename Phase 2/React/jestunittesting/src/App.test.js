import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
//Enzyme is used to create an instance
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe("App counter component", function() {
  it("starts with a count 0",() => {
    //create an instance of App
    var wrapperInstance= shallow(<App />);
    var count = wrapperInstance.state().count;
    expect(count).toEqual(0);
  });
  it("can increment count on click of button",() => {
    //create an instance of App
    var wrapperInstance = shallow(<App />);
    var incrementBtn = wrapperInstance.find('button');
    incrementBtn.simulate('click');
    var pText= wrapperInstance.find('p').text();
    expect(pText).toEqual('Count : 1');
  });
  it("takes snapshot",()=>{
    //create an instance of App
    var tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
    //jest --updateSnapshot to update snapshot
  });
});
