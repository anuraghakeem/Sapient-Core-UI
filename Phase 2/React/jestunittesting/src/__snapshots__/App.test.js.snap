// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App counter component takes snapshot 1`] = `
<div
  className="App"
>
  <p>
    Count : 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment Count
  </button>
</div>
`;
